lib/commons/dom/get-visible-child-text-rects.js

```diff
@@ -39,8 +39,14 @@ const getVisibleChildTextRects = memoize(
      * @see https://github.com/dequelabs/axe-core/issues/2178
      * @see https://github.com/dequelabs/axe-core/issues/2483
      * @see https://github.com/dequelabs/axe-core/issues/2681
+     *
+     * also need to resize the nodeRect to fit within the bounds of any overflow: hidden ancestors.
+     *
+     * @see https://github.com/dequelabs/axe-core/issues/4253
      */
-    return clientRects.length ? clientRects : [nodeRect];
+    return clientRects.length
+      ? clientRects
+      : filterHiddenRects([nodeRect], overflowHiddenNodes);
```

Comment:
Decided to leave this in as a failsafe though it technically shouldn't be needed anymore